### PR TITLE
Add support to deploy and test from deploy VM node

### PR DESCRIPTION
**Pull request by mkam**
Adds the VM parameter to multiple method so that we can run tasks on the newly added deploy node of the multi-node AIO environment.

I used named parameters instead of `Map args` with this card in mind: https://github.com/rcbops/u-suk-dev/issues/1531

Connects https://github.com/rcbops/u-suk-dev/issues/1578

### CODE REVIEW

**Comment by mkam** [comment]
Deployed and tested Mitaka: https://rpc.jenkins.cit.rackspace.net/job/OnMetal-Multi-Node-AIO/123/

The maas failures and tempest failures should be unrelated to the deployment process.

**Comment by mkam** [comment]
Build against master had a successful deployment, but also had tempest failures. All other stages passed, though.

https://rpc.jenkins.cit.rackspace.net/view/Multi-Node%20AIO/job/OnMetal-Multi-Node-AIO/127/

File: pipeline_steps/holland.groovy [hunk 1]
```diff
@@ -10,10 +10,18 @@ def holland(){
...
+      } else {
+        sh "scp -o StrictHostKeyChecking=no -p rpc-gating/playbooks/test_holland.yml deploy1:/opt/rpc-openstack/rpcd/playbooks"
+      }
+
       common.openstack_ansible(
+        vm: vm,
         playbook: "test_holland.yml",
-        path: "/opt/rpc-openstack/rpcd/playbooks"
+        path: "/opt/rpc-openstack/rpcd/playbooks",
+        enviornment_vars: ["ANSIBLE_CACHE_PLUGIN=memory"]
```
Comment (mattt416) [inline pipeline_steps/holland.groovy]: Typo on `environment_vars`.  Since this isn't getting passed in correctly, is it actually needed?

Comment (mkam) [inline pipeline_steps/holland.groovy]: ...I might have copied and pasted my typo everywhere! I'll update them.

File: rpc_jobs/lem_multi_node_aio.yml [hunk 2]
```diff
@@ -128,16 +129,13 @@
...
-                git branch: env.RPC_GATING_BRANCH, url: env.RPC_GATING_REPO
-              }}
-              maas.prepare(instance_name: instance_name)
-            }}
+            maas.prepare(instance_name: instance_name)
             maas.deploy()
             maas.verify()
-            tempest.tempest(vm: "infra1")
-            kibana.kibana()
+            tempest.tempest(vm: "infra1", "deploy1")
```
Comment (mattt416) [inline rpc_jobs/lem_multi_node_aio.yml]: Can you remove reference to `vm` here?

File: pipeline_steps/holland.groovy [hunk 3]
```diff
@@ -10,10 +10,18 @@ def holland(){
       / * recent versions of openstack-ansible allow execution of playbooks
         * from any directory, but that doesn't work all the way back to liberty
         */
-      sh """cp rpc-gating/playbooks/test_holland.yml /opt/rpc-openstack/rpcd/playbooks"""
+      
+      if (vm == null) {
+        sh "cp rpc-gating/playbooks/test_holland.yml /opt/rpc-openstack/rpcd/playbooks"
+      } else {
+        sh "scp -o StrictHostKeyChecking=no -p rpc-gating/playbooks/test_holland.yml deploy1:/opt/rpc-openstack/rpcd/playbooks"
```
Comment (mattt416) [inline pipeline_steps/holland.groovy]: We hardcode `deploy1` here, rather than using the `vm` var.

**Comment by mattt416** [comment]
LGTM, just testing this [here](https://rpc.jenkins.cit.rackspace.net/job/RPC-AIO_mitaka-swift-periodic/51/) quickly before merging.

**Comment by mattt416** [comment]
The above test failed on the actual cleanup, but otherwise looked good.

**Comment by mattt416** [comment]
@mkam looks like this one needs rebasing -- I'll merge this once that's done.

**Pull request merged**